Haskell/Haskell.sublime-syntax

```diff
@@ -5,68 +5,140 @@ name: Haskell
 file_extensions:
   - hs
 scope: source.haskell
+
+variables:
+  # In case this regex seems overly general,
+  # note that Haskell permits the definition of new operators
+  # which can be nearly any string of punctuation characters,
+  # such as $%^&*.
+  operator_infix: '[*|!%$?~+:\-.=</>\\]+'
```

Comment:
With extensions, Haskell also permits arbitrary unicode operators, so this regex is actually overly-specific.

Reply:
Not exactly arbitrary. I intended to go through the list I linked in OP and add those to this variable.